modules/ROOT/pages/bat-example-test-suite.adoc

```diff
@@ -0,0 +1,146 @@
+
+=== The `reporters` Section of the `bat.yaml` File
+
+This section is for configuring various tools or third-party applications to report test failures. In this example, a failure of the `HelloWorld.dwl` test prompts the BAT CLI to write a report of the failure to the file `/tmp/HTML.html`. For information about setting up reports, see xref:bat-reporting-task.adoc[Configure and View Test Reports].
+
+=== The `secrets` Section of the `bat.yaml` file
+
+If you plan to run a test suite from a private location, you can use the BAT CLI's integration with Anypoint Secrets Manager to use shared secrets in your tests.
+
+=== The `config` Folder
+
+You can store variables in the files in the `config` folder. For example, let's say you're developing an API and you deploy it to two different environments: `dev` and `qa`. The endpoint url for the first is `dev.myapi.com` and for the second `qa.myapi.com`. The API should behave the same in both environments.
+
+You want to test the API in both environments. However, instead of coding two tests, one for the deployment of the API in `dev` and one for the deployment of the API in `qa`, you can create a config file for each environment. In each config file, you can place a variable named `url` that holds that endpoint URL for the corresponding environment:
+
+.The content of `dev.dwl`, the config file for the `dev` environment
+----
+{
+url: "dev.myapi.com"
+}
+----
+
+.The content of `qa.dwl`, the config file for the `qa` environment
+----
+{
+url: "qa.myapi.com"
+}
+----
+
+In your test, you can use the `url` variable by placing it within this syntax: `$(config.variable)`, where _variable_ is the name of the variable that you want to use. Here is an example test that uses the `url` variable in a `GET` request to test the assertion that the response code must be 200:
+[linenums]
+----
+bat::BDD
+import * from bat::Assertions
+---
+suite("Hello world suite") in [
+  it must 'answer 200' in [
+    GET `$(config.url)` with {} assert [
+      $.response.status mustEqual 200
+  ]
+]
+----
+
+When you run the test, you can use the `--config` parameter to specify which config file to use. (If you upload your test suite as a monitor to API Functional Monitoring in Anypoint Platform, you can specify the configuration to use when you create a schedule for the monitor.)
+
+Use JSON for the format of your config files, and `.dwl` as the file extension. You can put as many variables as you need in each file.
+
+=== The `tests` folder
+
+In the example test suite, the test -- `HelloWorld.dwl` -- is in a folder named `tests`. As explained in the section above about the `bat.yaml` file in this example, the `files` section in the `bat.yaml` file points to the tests in a test suite.
```

Comment:
GLOBALLY replace dashes (and their keyboard replacements using hyphens and spaces).
In this text, you don't need anything.
Also, GLOBALLY remove references to content "above" or "below."
```suggestion
In the example test suite, the test `HelloWorld.dwl` is in a folder named `tests`. As explained in the section  about the `bat.yaml` file in this example, the `files` section in the `bat.yaml` file points to the tests in a test suite.
```

Reply:
Done.